astr303a.py

```diff
@@ -0,0 +1,571 @@
+import numpy as np
+import pylab as py
+from scipy import optimize
+from matplotlib.patches import Ellipse
+import subprocess
+import sys
+import re
+import os
+import io
+import scipy
+import shutil
+from matplotlib import pylab
+from matplotlib import rc
+from matplotlib import pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+
+#Create a pdf file that all figures will be saved to.
+global pp
+pp = PdfPages('GALAXIES.pdf')
+
+#filename of text file containing SDSS-DR13 data.
+filename = 'specobjdataset.txt'
+
+#read data into memory and organize into an array
+def readData(filename):
```

Comment:
This entire function can be replaced more robustly with 2-3 lines of Python standard library code.